hls-test-utils/src/Test/Hls.hs

```diff
@@ -134,3 +137,15 @@ runSessionWithServer' plugin conf sconf caps root s = withLock lock $ keepCurren
       (t, _) <- duration $ cancel server
       putStrLn $ "Finishing canceling (took " <> showDuration t <> "s)"
   pure x
+
+-- | Wait for all progress to be done
+-- Needs at least one progress done notification to return
+waitForProgressDone :: Session ()
+waitForProgressDone = loop
+  where
+    loop = do
+      ~() <- skipManyTill anyMessage $ satisfyMaybe $ \case
```

Comment:
What does `~()` mean? A lazy pattern match on `()`? How is that different from just ignoring the result?

Reply:
I have no idea - it's part of the code that was already there. I can't think of any other meaning

Reply:
Maybe it enforces the type in a way that used to be ambiguous? I'd be OK with `_ <-` or `() <-`.